### PR TITLE
Voting updated

### PR DESCRIPTION
**Pull request by ayush2843**

### CODE REVIEW

File: contracts/Pool/Pool.sol [hunk 1]
```diff
@@ -554,6 +557,55 @@ contract Pool is ERC20PresetMinterPauserUpgradeable, IPool {
...
+    function requestExtension() external isPoolActive OnlyBorrower
+    {
+        uint256 _extensionVoteEndTime = extensionVoteEndTime;
+        require(
+            block.timestamp > _extensionVoteEndTime,
+            "Pool::requestExtension - Extension requested already"
+        );
+
+        // This check is required so that borrower doesn't ask for more extension if previously an extension is already granted
+        require(periodWhenExtensionIsPassed > noOfRepaymentIntervals,"Pool::requestExtension: you have already been given an extension,No more extension");
```
Comment (prateekreddy) [inline contracts/Pool/Pool.sol:569]: If this is non zero that should be fine ? periodWhenExtensionIsPassed

File: contracts/Pool/Pool.sol [hunk 2]
```diff
@@ -554,6 +557,55 @@ contract Pool is ERC20PresetMinterPauserUpgradeable, IPool {
...
+            "Pool::voteOnExtension - you have already voted"
+        );
+        _lastVoteTime = block.timestamp;
+        totalExtensionSupport = totalExtensionSupport.add(balanceOf(msg.sender));
+        uint256 _votingPassRatio = IPoolFactory(PoolFactory).votingPassRatio();
+        lenders[msg.sender].lastVoteTime = _lastVoteTime;
+        emit lenderVoted(msg.sender,totalExtensionSupport,_lastVoteTime);
+        
+        if (((totalExtensionSupport)) >= (totalSupply().mul(_votingPassRatio)).div(100000000)) {
+            periodWhenExtensionIsPassed = calculateCurrentPeriod();
```
Comment (prateekreddy) [inline contracts/Pool/Pool.sol:603]: Is gracePeriod considered as part of the currentPeriod only, or does it overflow into the next period

Comment (prateekreddy) [reply contracts/Pool/Pool.sol:603]: If extension is approved in grace Period, the currentPeriod should be incremented right ? Also how is grace Penality being detected ?

File: contracts/Pool/Pool.sol (hunk 1 above)
Comment (ayush2843) [reply contracts/Pool/Pool.sol:569]: Zero will not work as voting can be passed in period zero also.

**Review by prateekreddy — APPROVED** [review]

**Pull request merged**